kubernetes-client-openapi/src/main/java/io/micronaut/kubernetes/client/openapi/config/KubeConfig.java

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.kubernetes.client.openapi.config;
+
+import io.micronaut.core.util.CollectionUtils;
+import io.micronaut.core.util.StringUtils;
+import io.micronaut.kubernetes.client.openapi.config.model.AuthInfo;
+import io.micronaut.kubernetes.client.openapi.config.model.Cluster;
+import io.micronaut.kubernetes.client.openapi.config.model.Context;
+import io.micronaut.kubernetes.client.openapi.config.model.ExecConfig;
+import io.micronaut.kubernetes.client.openapi.config.model.ExecEnvVar;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Holder for data loaded from the kube config file.
+ */
+public class KubeConfig {
```

Comment:
```suggestion
public final class KubeConfig {
```

Reply:
is this mean to be public API?

Reply:
This is just a holder for data read from the kube config file.

Reply:
ok make it package private and internal if possible